astropy/units/format/base.py

```diff
@@ -144,7 +165,7 @@ def to_string(cls, unit, *, fraction=True):
         return s
 
     @classmethod
-    def parse(cls, s):
```

Comment:
tangent, but I don't get why this method exists. Is it conceptually an abstract method ?

Reply:
If it is, we should decorate it as such.

Reply:
agreed

Reply:
`Base` is not an abstract class, but apparently it is supposed to be:
https://github.com/astropy/astropy/blob/cec24e83efea38ee18b80b7b34693a209d07d595/astropy/units/format/base.py#L4-L7
It looks like `parse()` should indeed be an abstract method, but this is not the right pull request for making that change.

Reply:
Note that these are singleton classes which do not get instantiated, so abstract methods etc are a bit different.

Reply:
This discussion is going beyond the scope of this pull request, but that being said I don't see why this couldn't be an abstract class method.

Reply:
Agreed. Let's move this to an Issue.

Reply:
https://github.com/astropy/astropy/issues/16374